### PR TITLE
fix(stack-view): buttons for screenreaders

### LINKED ISSUE #3567: Stack View button has illogical sequence for screen readers
## Describe the bug

The interstitial content in the stack view presents an illogical reading sequence for screen reader users. When screen reader users hear a button announce an expanded or collapsed state they will think that the button controls the readability of upcoming content. 

Discovering content related to the stack-block-label button immediately after hearing that it is in a collapsed state may be confusing. While this is a low severity issue when there is a small amount of interstitial content, the severity will increase in relation to the amount of interstitial content.

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Using a screen reader go to Stackview example
- Read one of the collapsed stack-block-label buttons
- Observe that you read the value of the stack's key value pair immediately after reading the collapsed button. 

## Expected behavior

The Stack view's implementation should combine the stack-block-label and the interstitial content in the first stack-block-content element into a single control, specifically a button. The stack's key value pair can be used to provide an accessible name and description for this button. This can be done with aria-label or aria-labelledby and aria-describedby.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._

### PR DESCRIPTION
**Pull request by hippee-lee**
This fix addresses an a11y issue with stackview buttons. It consoledates the label and content child components into the button.
This allows screen readers to announce both when the button is focused. Styles have been updated to preserve visual design.
This closes #3567.

Signed-off-by: Matt Hippely <mhippely@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?
Change in stack-view and stack-block component templates. The updates allows screen readers to access both the label and the content projected by consumers. 

* [x] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?
Screen readers only announce both the label projected by consumers.

Issue Number: N/A

## What is the new behavior?
Screen readers announce both the label and the content projected by consumers. 

## Does this PR introduce a breaking change?

* [x] Yes (@clr/ui consumers need to add an extra element with the `stack-view-key` class. 
* [ ] No

### CODE REVIEW

File: src/clr-angular/data/stack-view/_stack-view.clarity.scss [hunk 1]
```diff
@@ -150,6 +149,14 @@ $styledInputs: 'input[type=text], input[type=password], input[type=number], inpu
       );
     }
 
+    .stack-block-content {
+      background-color: inherit;
```
Comment (mathisscott) [inline src/clr-angular/data/stack-view/_stack-view.clarity.scss]: 👍

File: src/clr-angular/data/stack-view/stack-block.ts [hunk 2]
```diff
@@ -139,6 +145,7 @@ export class ClrStackBlock implements OnInit {
     if (!this.expandable) {
       return null;
     } else {
+      console.log(this.expanded);
```
Comment (mathisscott) [inline src/clr-angular/data/stack-view/stack-block.ts]: Do we want this `console.log` here?

Comment (hippee-lee) [reply src/clr-angular/data/stack-view/stack-block.ts]: nope. I'll remove it.

**Comment by Jinnie** [comment]
Why removing the DL context? Curious.

**Comment by chlane** [comment]
> Why removing the DL context? Curious.

I advised Matt to follow https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure. The definition list implementation actually creates the interstitial content problem.

**Pull request merged**